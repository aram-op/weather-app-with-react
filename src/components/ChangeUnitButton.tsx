import React from 'react';
import '../styles/change-unit-button.css';

function ChangeUnitButton({onUnitChange} : {onUnitChange: (e: React.MouseEvent) => void}) {
    return(
        <button
            className='unit-change-btn'
            onClick={onUnitChange}
        >
            Change Unit
        </button>
    );
}

export default ChangeUnitButton;