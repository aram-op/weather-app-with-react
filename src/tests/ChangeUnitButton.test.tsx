import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import ChangeUnitButton from '../components/ChangeUnitButton';

describe('ChangeUnitButton', () => {

    it('should render correctly', () => {
        const { asFragment } = render(<ChangeUnitButton onUnitChange={jest.fn()} />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('should call onUnitChange when the button is clicked', () => {
        const onUnitChangeMock = jest.fn();

        render(<ChangeUnitButton onUnitChange={onUnitChangeMock} />);

        const button = screen.getByText('Change Unit');

        fireEvent.click(button);

        expect(onUnitChangeMock).toHaveBeenCalledTimes(1);
    });
});
