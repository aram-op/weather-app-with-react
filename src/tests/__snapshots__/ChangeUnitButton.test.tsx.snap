// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeUnitButton renders correctly 1`] = `
<DocumentFragment>
  <button
    class="unit-change-btn"
  >
    Change Unit
  </button>
</DocumentFragment>
`;

exports[`ChangeUnitButton should render correctly 1`] = `
<DocumentFragment>
  <button
    class="unit-change-btn"
  >
    Change Unit
  </button>
</DocumentFragment>
`;
